configure.ac

```diff
@@ -216,6 +218,48 @@ AC_CHECK_HEADER([security/_pam_types.h],
 AC_CHECK_HEADER([security/pam_constants.h],
                 [AC_DEFINE([HAVE_PAM_CONSTANTS_H], 1, [Using OpenPAM], [])])
 
+# Find imlib2
+case "$with_imlib2" in
+    no) AC_MSG_NOTICE([imlib2 will not be supported])
+        use_imlib2=no
+        ;;
+    '' | auto)
+        PKG_CHECK_MODULES([IMLIB2], [imlib2 >= 1.4.10],
+            [use_imlib2=yes],
+            [use_imlib2=no])
+        ;;
+    yes)
+        PKG_CHECK_MODULES([IMLIB2], [imlib2 >= 1.4.10],
+            [use_imlib2=yes],
+            [AC_MSG_ERROR([please install libimlib2-dev or imlib2-devel])])
+        ;;
+    *)
+        AC_MSG_CHECKING([for imlib2 in $with_imlib2])
+        if test -d $with_imlib2/lib; then
+            IMLIB2_LIBS="-L $with_imlib2/lib -lImlib2"
+        elif test -d $with_imlib2/lib64; then
+            IMLIB2_LIBS="-L $with_imlib2/lib64 -lImlib2"
+        else
+            AC_MSG_RESULT([no])
+            AC_MSG_ERROR([Can't find libImlib2 in $with_imlib2])
+        fi
+
+        if test -f $with_imlib2/include/Imlib2.h; then
+            IMLIB2_CFLAGS="-I $with_imlib2/include"
+        else
+            AC_MSG_RESULT([no])
+            AC_MSG_ERROR([Can't find $with_imlib2/include/Imlib2.h])
+        fi
+        AC_MSG_RESULT([yes])
+        AC_SUBST([IMLIB2_LIBS])
+        AC_SUBST([IMLIB2_CFLAGS])
+        use_imlib2=yes
+esac
+
```

Comment:
Auto-detection is generally very nice but autodetected libraries will not appear in `xrdp -v`. I want an easy way to tell users if your compiled xrdp supports png/jpg images on the login screen. `ldd /usr/sbin/xrdp` sounds a little bit guru's way.

If we introduce more auto-detection, we might need to invent a new way to show which libraries/features compiled with.

Reply:
That's a really good point - I hadn't thought of that but it's a feature I use a lot when triaging problems. I'll remove the auto-detect.